core/components/menu/menuitem.js

```diff
@@ -25,121 +24,108 @@ goog.require('Blockly.utils.object');
  *     the item.
  * @param {string=} opt_value Data/model associated with the menu item.
  * @constructor
- * @extends {Blockly.Component}
  */
 Blockly.MenuItem = function(content, opt_value) {
-  Blockly.Component.call(this);
+  /**
+   * Human-readable text of this menu item.
+   * @type {string}
+   * @private
+   */
+  this.content_ = content;
 
-  this.setContentInternal(content);
-  this.setValue(opt_value);
+  /**
+   * Machine-readable value of this menu item.
+   * @type {string|undefined}
+   * @private
+   */
+  this.value_ = opt_value;
 
   /**
+   * Is the menu item clickable, as opposed to greyed-out.
    * @type {boolean}
    * @private
    */
   this.enabled_ = true;
 
   /**
-   * @type {Blockly.MenuItem}
+   * The DOM element for the menu item.
+   * @type {?Element}
    * @private
    */
-  this.previousSibling_;
+  this.element_ = null;
 
   /**
-   * @type {Blockly.MenuItem}
-   * @private
+   * Whether the menu item is rendered right-to-left.
+   * @type {boolean}
+   * @protected
    */
-  this.nextSibling_;
+  this.rightToLeft_ = false;
```

Comment:
Same here (and this doesn't need to be done in this PR, since its not a regression), but we should add definitions for: 
``this.checked_``, ``this.checkable_``, ``this.roleName``, and ``this.actionHandler_``.

Reply:
Done.